pkg/client/cache/index.go

```diff
@@ -28,6 +28,8 @@ type Indexer interface {
 	Store
 	// Retrieve list of objects that match on the named indexing function
 	Index(indexName string, obj interface{}) ([]interface{}, error)
+	// Retrieve list of all index names
+	ListIndexNames(name string) []string
```

Comment:
This should not take a ```name``` argument.  The idea is to basically give all the keys in the type ```Indexers``` so I do not see why a name argument is provided.